session/manager.go

```diff
@@ -221,13 +224,23 @@ func (manager *Manager) Acknowledge(consumerID identity.Identity, sessionID stri
 	return nil
 }
 
+func (manager *Manager) clearStaleSession(consumerID identity.Identity, serviceType string) {
+	sessionID, ok := manager.sessionStorage.FindBy(FindOpts{
+		Peer:        &consumerID,
+		ServiceType: serviceType,
+	})
+	if ok {
+		log.Info().Msgf("Cleaning stale session %s for %s consumer", sessionID, consumerID.Address)
+		go manager.Destroy(consumerID, string(sessionID))
```

Comment:
This will triggers storage traversal 2 times, reuse found `session` object

Reply:
I'm not sure we want to reuse `Destroy()` here, it's logic has different purpose

Reply:
I do not understand this. 
What is storage traversal? 
What logic has `Destroy` except the destroying session?

Reply:
@Waldz Please answer comment. Thanks.

Reply:
Session storage traversal is happening 2 times with session find methods.

`Destroy` purpose is for public contract. It does 2 things: a) finds session + b) cleans found session.
Your new use-case already has found session and you need just b) clean found session. So:
- either make needed part resuable in `destroySession(session Session)`
- either just call cleaning for found sessions:
```
manager.sessionStorage.Remove(ID(sessionID))
close(session.done)
```

Reply:
Extracted `destroySession` from `Destroy`.